src/v2/providers/schedule.ts

```diff
@@ -0,0 +1,198 @@
+
+  /** The minimum time to wait before retying. */
+  minBackoffDuration?: string;
+
+  /** The maximum time to wait before retrying. */
+  maxBackoffDuration?: string;
+
+  /** The time between will double max doublings times. */
+  maxDoublings?: number;
+}
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param schedule - The schedule, in Unix Crontab or AppEngine syntax.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  schedule: string,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction;
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param options - Options to set on scheduled functions.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  options: ScheduleOptions,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction;
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param args - Either a schedule or an object containing function options.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  args: string | ScheduleOptions,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction {
+  const separatedOpts = getOpts(args);
+
+  const func: any = async (
+    req: express.Request,
+    res: express.Response
+  ): Promise<any> => {
+    try {
+      await handler(req);
+    } catch (err) {
+      logger.warn((err as Error).message);
+    }
+
+    res.setHeader('Content-Type', 'application/json');
+    res.status(200).send();
+  };
+  func.run = handler;
+
+  const baseOptsEndpoint = options.optionsToEndpoint(
+    options.getGlobalOptions()
+  );
+  const specificOptsEndpoint = options.optionsToEndpoint(separatedOpts.opts);
+
+  const ep: ManifestEndpoint = {
+    platform: 'gcfv2',
+    ...baseOptsEndpoint,
+    ...specificOptsEndpoint,
+    labels: {
+      ...baseOptsEndpoint?.labels,
+      ...specificOptsEndpoint?.labels,
+    },
+    scheduleTrigger: {
+      schedule: separatedOpts.schedule,
+      retryConfig: {},
+    },
+  };
+  copyIfPresent(ep.scheduleTrigger, separatedOpts, 'timeZone');
+  copyIfPresent(
+    ep.scheduleTrigger.retryConfig,
+    separatedOpts,
+    'retryCount',
+    'maxRetryDuration',
+    'minBackoffDuration',
+    'maxBackoffDuration',
+    'maxDoublings'
+  );
+  // TODO(colerogers): add invoker to scheduleTrigger
+  // and the container contract if we want to
+  // support enterprise customers and change the
+  // behavior of using the default compute service agent as
+  // the function invoker from cloud scheduler.
```

Comment:
q: is this a quick follow up or something we want to punt until we have explicit request?

(I imagine it will be a pita to test this internally :()

Reply:
I was leaving this a follow up since I didn't want to wait for another API review since this is modifying the container contact. But I can block on that if you think we should launch with custom invoker support

Reply:
sgtm. Maybe we can create a ticket/github issue instead. I know that is as good as throwing a ticket down a black hole, but I hope it's more discoverable in the future.

Reply:
SG, created an internal issue